source/core/sdirk_stage_data.cc

```diff
@@ -0,0 +1,35 @@
+#include <core/sdirk_stage_data.h>
+
+#include <deal.II/base/logstream.h>
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <iomanip>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+using namespace dealii;
+
+SDIRKStageData
+sdirk_stage_data(const FullMatrix<double>  &butcher_table,
+                 const std::vector<double> &c,
+                 const std::vector<double> &b,
+                 const unsigned int         stage_i)
+{
+  const unsigned int n_stages = b.size();
```

Comment:
I would assert that c.size is equal to b.size no?

Reply:
I would also AssertThrow for the size of the butcher table, making sure that everyone is consistent.